hera_qm/utils.py

```diff
@@ -691,62 +703,72 @@ def apply_yaml_flags(uv, a_priori_flag_yaml, lat_lon_alt_degrees=None, telescope
-            antnum = ant[0]
+            if issubclass(uv.__class__, UVData) or (isinstance(uv, UVFlag) and uv.type == 'baseline'):
+                uv.flag_array[:, 0, flagged_channels, :] = True
+            elif issubclass(uv.__class__, UVCal) or (isinstance(uv, UVFlag) and uv.type == 'antenna'):
+                uv.flag_array[:, 0, flagged_channels, :, :] = True
+            elif isinstance(uv, UVFlag) and uv.type == 'waterfall':
+                uv.flag_array[:, flagged_channels] = True
+    if flag_times:
+        # now do times.
+        # get the integrations to flag
+        flagged_integrations = metrics_io.read_a_priori_int_flags(a_priori_flag_yaml, lsts=lst_array, times=time_array)
+        # ony select integrations less then Ntimes
+        if np.any(flagged_integrations >= uv.Ntimes):
+            warnings.warn("Flagged integrations were provided that exceed the maximum integration index. These flags are being dropped!")
+        if np.any(flagged_integrations < 0):
+            warnings.warn("Flagged integrations were provided with a negative integration index. These flags are being dropped!")
+        flagged_integrations = flagged_integrations[(flagged_integrations>=0) & (flagged_integrations<=uv.Ntimes)]
+        if len(flagged_integrations) > 0:
+            flagged_times = time_array[flagged_integrations]
+            for time in flagged_times:
+                if issubclass(uv.__class__, UVData) or (isinstance(uv, UVFlag) and uv.type == 'baseline'):
+                    uv.flag_array[np.isclose(uv.time_array, time), :, :, :] = True
+                elif issubclass(uv.__class__, UVCal) or (isinstance(uv, UVFlag) and uv.type == 'antenna'):
+                    uv.flag_array[:, :, :, np.isclose(uv.time_array, time), :] = True
+            if isinstance(uv, UVFlag) and uv.type == 'waterfall':
+                uv.flag_array[flagged_integrations, :] = True
+
+    if flag_ants:
+        # now do antennas.
+        flagged_ants = metrics_io.read_a_priori_ant_flags(a_priori_flag_yaml, ant_indices_only=ant_indices_only,
+                                                          by_ant_pol=by_ant_pol, ant_pols=ant_pols)
+        npols = uv.flag_array.shape[-1]
         if issubclass(uv.__class__, UVData):
-            blt_selection = np.logical_or(uv.ant_1_array == antnum, uv.ant_2_array == antnum)
-            if np.any(blt_selection):
-                for bltind in np.where(blt_selection)[0]:
-                    uv.flag_array[bltind, :, :, pol_selection] = True
+            pol_array = uv.polarization_array
         elif issubclass(uv.__class__, UVCal):
-            ant_selection = uv.ant_array == antnum
-            if np.any(ant_selection):
-                for antind in np.where(ant_selection)[0]:
-                    uv.flag_array[antind, :, :, :, pol_selection] =True
+            pol_array = uv.jones_array
+        elif issubclass(uv.__class__, UVFlag):
+            pol_array = uv.polarization_array
+
```

Comment:
I don't think you quite finished here: You need to handle the cases where it's a baseline time, an antenna type, and a waterfall type (which should do nothing).

Reply:
It seems that the polarization_array is always there, even if the uvflag is antenna type (despite the fact that uvcal objects use jones_array instead of polarization_array). Here is the line where the assignment happens.

https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/eedb1e10ef2af8e4a019e7df2abed012d23c6471/pyuvdata/uvflag/uvflag.py#L3042



 If the type is a waterfall, the the polarization array contains a string so that no pols will come up as being contained in the array so nothing should happen.